x-pack/plugins/enterprise_search/common/types/indices.ts

```diff
@@ -16,8 +16,14 @@ import { Connector } from './connectors';
 import { Crawler } from './crawler';
 
 export interface ElasticsearchIndex {
+  alias: boolean;
   health?: HealthStatus;
   name: IndexName;
+  privileges: {
```

Comment:
Missed this in my initial review, but having these as optional isn't great--it'll really hinder us down the road with unnecessary undefined checks. Better to have these always be defined, and then returning them with sensible defaults in the return function, eg:
```
privileges : {
  read: false;
  manage: false
}
```

Reply:
Wow, I had no idea defaults were a thing!

Reply:
Hmmm, I still get Typescript errors for this:
```
Type 'SecurityHasPrivilegesPrivileges' is missing the following properties from type '{ read: false; manage: false; }': read, manage
Type 'SecurityHasPrivilegesPrivileges' is not assignable to type '{ read: false; manage: false; }'.	
```

Reply:
I think maybe I understand what you meant now. I kept the interface as:
```
privileges: {
  read: boolean;
  manage: boolean;
}
```
and then in the assignment part in the function, I did:
```
privileges: {
  read: false,
  manage: false,
  ...indexData[indexName],
}
```